### PR TITLE
Add grpc health check. (#236)

### PR DESCRIPTION
**Pull request by moeHaydar**
Added a grpc health check with the following configuration: `grpc.server.health.enabled`

### CODE REVIEW

**Review by ilopmar — CHANGES_REQUESTED** [review]
Can you please add some tests?

**Comment by moeHaydar** [comment]
> Can you please add some tests?

Sure. I added a couple.

File: src/main/docs/guide/server.adoc [hunk 1]
```diff
@@ -108,6 +108,30 @@ and then the order will be reversed when sending the response message.
 Request -> 1 -> 2 -> 3 -> business logic -> 3 -> 2 -> 1 -> Response
 ----
 
+=== Health checks
+If the management dependency ("io.micronaut:micronaut-management") is added, then Micronaut’s Health Endpoint can be used to expose the health status of the Grpc server.
+
+For example if grpc is running the /health endpoint will return:
```
Comment (graemerocher) [inline src/main/docs/guide/server.adoc]: Could you change the references to gRPC to be consistently cased. I believe the correct casing is `gRPC`.

Comment (moeHaydar) [reply src/main/docs/guide/server.adoc]: Fixed

Comment (ilopmar) [reply src/main/docs/guide/server.adoc]: Done here. I'll create a new PR to change it in all the docs.

**Review by ilopmar — APPROVED** [review]

**Comment by ilopmar** [comment]
Thanks for the contribution!

**Pull request merged**